### PR TITLE
Define projected skill runtime contract and repo-root fallback

### PR DESCRIPTION
**Pull request by shaug**
# Summary

- Define the shared projected-skill runtime contract so bootstrap policy is explicit before the runtime repair changes land.

# Changes

- Add `ProjectedRuntimeMode`, `ProjectedRuntimeContract`, and `projected_runtime_contract()` in `src/atelier/runtime_env.py` to define supported modes, provenance selection rules, inherited `PYTHONPATH` safety rules, and deterministic `repo_root = None` behavior.
- Route the shared projected bootstrap helper through that contract and document the policy in `docs/projected-skill-runtime-contract.md`.
- Add focused runtime contract tests covering repo-source preference and `repo_root = None` no-reexec behavior.

# Testing

- `just lint`
- `just test`

## Tickets
- None

# Risks / Rollout

- This changeset defines policy and wires the shared bootstrap to that contract, but it intentionally does not change the later runtime sanitization repair behavior.

# Notes

- Later projected-runtime fixes can now implement against one shared contract instead of re-deciding bootstrap policy per skill.

### CODE REVIEW

File: src/atelier/runtime_env.py [hunk 1]
```diff
@@ -50,6 +51,102 @@ class _ProjectedRuntimeProvenanceIssue:
...
+        ),
+        preferred_mode=preferred_mode,
+        repo_root_behavior=repo_root_behavior,
+        provenance_selection_rules=(
+            "Resolve repo provenance explicitly from --repo-dir, the local "
+            "./worktree link, projected repo env hints, then cwd/script "
+            "ancestry.",
+            "Use repo-source mode only after bootstrap proves a checkout with "
+            "src/atelier is available for the projected script.",
+            "When repo_root is None, remain in active-interpreter mode and "
```
[the remark this answers is not in the document]
Comment (shaug) [reply src/atelier/runtime_env.py]: Updated this slice so it no longer encodes unconditional inherited `PYTHONPATH` discard. The shared contract now says bootstrap must clear or reduce `PYTHONPATH` to import roots already proven to belong to the selected runtime, with explicit separate rules for `active-interpreter` and `repo-source` modes. The matching doc and test updates are in `4c19ddb`.
